resources/assets/js/components/ApplicantProfile/Skills/SkillSearchResults.tsx

```diff
@@ -0,0 +1,199 @@
+    >
+      <button
+        aria-expanded="false"
+        data-h2-accordion-trigger
+        type="button"
+        tabIndex={0}
+      >
+        <span data-h2-accordion-trigger-label>
+          {intl.formatMessage(resultMessages.resultAccordionExpand)}
+        </span>
+        <span
+          aria-hidden="true"
+          data-h2-accordion-add-icon
+          data-h2-font-size="b(h4)"
+          data-h2-font-color="b(theme-1)"
+        >
+          +
+        </span>
+        <span
+          aria-hidden="true"
+          data-h2-accordion-remove-icon
+          data-h2-font-size="b(h4)"
+          data-h2-font-color="b(theme-1)"
+        >
+          -
+        </span>
+        <div data-h2-accordion-trigger-content>
+          <p>
+            <span>{localizeFieldNonNull(locale, item, "name")}</span>
+            {(item.isChecked || added) && (
+              <span data-h2-color="theme-1" data-h2-padding="b(left, 1)">
+                <i className="fas fa-check" />
+              </span>
+            )}
+          </p>
+        </div>
+      </button>
+      <div aria-hidden="true" data-h2-accordion-content>
+        <div data-h2-grid="b(middle, contained, padded, 0)">
+          <div data-h2-grid-item="b(2of3)">
+            <div data-h2-grid-content>
+              <p data-h2-focus>
+                {localizeFieldNonNull(locale, item, "description")}
+              </p>
+            </div>
+          </div>
+          <div data-h2-grid-item="b(1of3)" data-h2-align="b(right)">
+            <div data-h2-grid-content data-h2-padding="b(right, 1)">
+              {!added && !item.isChecked && (
+                <button
+                  data-h2-button="gray-9, pill, medium, outline"
+                  type="button"
+                  disabled={isSubmitting}
+                  onClick={() => {
+                    setIsSubmitting(true);
+                    handleAddSkill(item.id)
+                      .then(() => {
+                        setAdded(true);
+                      })
+                      .catch(() => {
+                        setIsSubmitting(false);
+                      });
+                  }}
+                >
+                  <span data-h2-button-label>
+                    {intl.formatMessage(resultMessages.resultAccordionAdd)}
+                  </span>
+                </button>
+              )}
+              {added && (
+                <p data-h2-padding="b(all, .5)" data-h2-font-weight="b(600)">
+                  {intl.formatMessage(resultMessages.resultAccordionAdded)}
+                </p>
+              )}
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export const SkillSearchResults: React.FunctionComponent<SkillSearchResultsProps> = ({
+  status,
+  results,
+  handleAddSkill,
+}: SkillSearchResultsProps): React.ReactElement => {
+  const intl = useIntl();
+  React.useEffect((): void => {
+    h2ComponentAccordionAddTriggerEvent();
+  }, [results]);
```

Comment:
I think it would be safer to leave the second arg out entirely, and have the effect run on every re-render.

Reply:
Actually, I believe this hook should be in the Item component, since if that component re-rendered, and the parent component didn't, we'd still need to re-attach the event listeners.

Reply:
@tristan-orourke I had previously tried both leaving out the second argument as well as moving the `useEffect` hook to then Item component (with and without the second argument). All of these variations resulted in the accordion not being able to toggle its expansion. Is there something else I'm missing here?

Reply:
Hmm, If that's the case, more likely I'm missing something here.